kasa/device_helpers.py

```diff
@@ -0,0 +1,89 @@
+"""Device creation by type."""
+
+from typing import Any, Dict, Optional, Tuple, Type
+
+from .aestransport import AesTransport
+from .connectionparams import ConnectionParameters
+from .credentials import Credentials
+from .exceptions import UnsupportedDeviceException
+from .iotprotocol import IotProtocol
+from .klaptransport import KlapTransport, TPlinkKlapTransportV2
+from .protocol import BaseTransport, TPLinkProtocol, TPLinkSmartHomeProtocol
+from .smartbulb import SmartBulb
+from .smartdevice import SmartDevice, SmartDeviceException
+from .smartdimmer import SmartDimmer
+from .smartlightstrip import SmartLightStrip
+from .smartplug import SmartPlug
+from .smartprotocol import SmartProtocol
+from .smartstrip import SmartStrip
+from .tapo import TapoBulb, TapoPlug
+
+
+def get_protocol_from_connection_parameters(
+    connection_params: ConnectionParameters,
+    host: str,
+    credentials: Optional[Credentials] = None,
+) -> Optional[TPLinkProtocol]:
+    """Return the protocol from the connection name."""
+    protocol_name = connection_params.tplink_device_type.value.split(".")[0]
+    protocol_transport_key = (
+        protocol_name + "." + connection_params.encryption_type.value
+    )
+    supported_device_protocols: dict[
+        str, Tuple[Type[TPLinkProtocol], Type[BaseTransport]]
+    ] = {
+        "IOT.XOR": (TPLinkSmartHomeProtocol, KlapTransport),
+        "IOT.KLAP": (IotProtocol, KlapTransport),
+        "SMART.AES": (SmartProtocol, AesTransport),
+        "SMART.KLAP": (SmartProtocol, TPlinkKlapTransportV2),
+    }
+    if protocol_transport_key not in supported_device_protocols:
+        return None
```

Comment:
This probably should raise an exception (or return a default combination?) instead of returning none.

Reply:
Done